Improve `find` command documentation in User Guide (#376)
AY2223S1-CS2103T-T12-4/tp: docs/UserGuide.md

### Finding patients: `find`

> Finds patients whose parameters contain the given keywords.

Reviewer: Perhaps it may be better to use more second-person language (I had updated it for conditions, tags, and other miscellaneous commands). 

```suggestion
You can find specific patients using the `find` command.
```

> Format: `find [KEYWORD]… [xyz/XYZ_KEYWORD]…`

Reviewer: Another change I made in the reformatted UG was to bold the command words in every "Format" line.
```suggestion
Format: **`find`**`[KEYWORD]… [xyz/XYZ_KEYWORD]…`
```

* `xyz/XYZ_KEYWORD` refers to a keyword for a specific parameter.
* There should be at least one parameter for the command.
* The search is case-insensitive. e.g `hans` will match `Hans`.
* The order of the keywords does not matter. e.g. `Hans Bo` will match `Bo Hans`.
* Partial words can be matched e.g. `Han` will match `Hans`.
* Patients matching at least one keyword in every parameter will be returned (i.e. AND search for different parameters, OR search for same parameter). In more details,

>     * At least one of the patient's details (name, phone, email, address, tag, condition, task description, or medication) must match with at least one `KEYWORD`.

Reviewer: Left out remark.
Not a mistake on your part, but I'm enforcing a certain order of the attributes for consistency, though I have forgotten to change some of them.
```suggestion
    * At least one of the patient's details (name, phone, email, address, tag, task description, condition, medication, or remark) must match with at least one `KEYWORD`.
```

>     * For every different parameter `xyz`, it must match at least one `XYZ_KEYWORD`.

Reviewer: It would refer to the word "parameter", which is defined in the glossary to be the part of a command after a prefix i.e. x/PARAMETER. I think it is more accurate to say a patient's details matches `XYZ_KEYWORD`.
```suggestion
    * For every different parameter `xyz`, the patient's corresponding details must match at least one `XYZ_KEYWORD`.
```

Examples:
* `find key n/John n/Betsy n/Charlie e/@example.com e/@u.nus.edu` returns patients who fulfill all conditions below:
    * The patient's name contains either `John` or `Betsy` or `Charlie`.

>     * The patient's email address must contains either `@example.com` or `@u.nus.edu`.

Reviewer: ```suggestion
    * The patient's email address must contain either `@example.com` or `@u.nus.edu`.
```
